src/v3/plugins/github/urlIdParse.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+// Given a sequence of regex parts that appear after the repo name and owner
+// in a GitHub url, create a regex to match them.
+function githubRegex(...regexParts) {
```

Comment:
Cool; this seems like a useful function.

The implementation, however, is not correct. The problem is that the
`RegExp` constructor is not homomorphic with respect to concatenation.
That is, there is a well-defined operation of regular expression
concatenation, which is what you’re trying to express, but this
operation is not equivalent to “return the `RegExp` whose source is the
string concatenation of the sources of the two `RegExp`s”. That is: if
we let `∙` denote regular expression concatenation and `++` denote
string concatenation, then it is not the case that

    RegExp (s1 ++ s2) == (RegExp s1) ∙ (RegExp s2),

but this is the assumption that your code is making.

As a concrete example, the result of

```js
// Find an issue or pull request!
const re = githubRegex(/issues|pull/, /(\d+)/);
```

is the expression `/https:…\/issues|pull\/(\d+)$/`, which is not
correct, because it matches a URL ending in `"/issues/foo/bar/baz"`.

Basically, the thing is vulnerable to “regular expression injection”.

We can fix this by surrounding the individual parts with non-capturing
groups, which I believe (but have not proven) to be sufficient. This
also suggests what I think is a significantly cleaner implementation:

```js
const parts = [
  /https:\/\/github.com/,
  githubNamePart,
  githubNamePart,
  ...regexParts,
];
const mainSource = parts.map((re) => `(?:${re.source})`).join("/");
return new RegExp(`^${mainSource}$`);
```

Accordingly, as you may expect, I would like to see this function
tested. :-)

Reply:
Good find. I removed this function as it's incorrect and I don't want to test it, given that I only needed it for 4 specific instances. PTAL.